### PR TITLE
Add rust support

### PR DESCRIPTION
**Pull request by purpleP**
## Checklist

- [ ] I have added [tests](https://github.com/pokey/cursorless-vscode/blob/main/docs/contributing/test-case-recorder.

That’s a work in progress so far. I have published it this early, so that no one else would waste their time.

### CODE REVIEW

**Comment by pokey** [comment]
👍 for opening early PR.  Please switch to the [adding a language template](https://raw.githubusercontent.com/cursorless-dev/cursorless-vscode/main/.github/PULL_REQUEST_TEMPLATE/new_programming_language.md) tho as it's got some helpful tips in there

File: src/languages/rust.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  string: "string_literal",
+  comment: ["line_comment", "block_comment"],
+  ifStatement: "if_expression",
+  list: "array_expression",
+  functionCall: "call_expression",
+  name: [
+      "let_declaration.identifier!",
+      "parameter.identifier!",
+  ],
+  type: trailingMatcher([
```
Comment (pokey) [inline src/languages/rust.ts:38]: I believe you want a `leadingMatcher` here, like we use for Python.  This way "chuck type" will remove the leading `:`

File: src/languages/rust.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  conditionMatcher,
+  trailingMatcher,
+} from "../util/nodeMatchers";
+import { NodeMatcherAlternative, ScopeType } from "../typings/Types";
+
+const STATEMENT_TYPES = [
+  "use_declaration",
+  "while_expression",
+  "loop_expression",
+  "block",
```
Comment (pokey) [inline src/languages/rust.ts:14]: You probably don't want to include `"block"`, as that will trap you between the curly brackets if you say "take state" within an if statement body

**Review by pokey — COMMENTED** [review]
Looks like a good start!  Definitely worth recording tests before you get much further.  It's _really_ easy.  I promise 😊

**Comment by trace-andreason** [comment]
@purpleP I had been working on adding rust support, and I've opened a PR here: https://github.com/cursorless-dev/cursorless-vscode/pull/546 so that you can compare your changes to mine

**Comment by pokey** [comment]
Ok closing in favour of #546, but I believe we're stealing some stuff from here so thanks very much for this PR! 😊

**Comment by purpleP** [comment]
Yes, guys, I have my hand full at the moment. Will help with testing If I can.

**Pull request closed without merging**